src/main/cpp/feedback/DriverFeedback.cpp

```diff
@@ -92,21 +98,22 @@ void DriverFeedback::UpdateLEDStates()
 
 void DriverFeedback::UpdateDiagnosticLEDs()
 {
-    /**
-    StateMgr *noteStateManager = MechanismConfigMgr::GetInstance()->GetCurrentConfig()->GetMechanism(MechanismTypes::NOTE_MANAGER);
-    auto noteMgr = noteStateManager != nullptr ? dynamic_cast<noteManagerGen *>(noteStateManager) : nullptr;
-    if (noteMgr != nullptr)
+    if (MechanismConfigMgr::GetInstance()->GetCurrentConfig() != nullptr)
     {
-        bool backintake = noteMgr->getbackIntakeSensor()->Get();
-        bool frontintake = noteMgr->getfrontIntakeSensor()->Get();
-        bool feeder = noteMgr->getfeederSensor()->Get();
-        bool launcher = noteMgr->getlauncherSensor()->Get();
-        bool placerin = noteMgr->getplacerInSensor()->Get();
-        bool placermid = noteMgr->getplacerMidSensor()->Get();
-        bool placerout = noteMgr->getplacerOutSensor()->Get();
-        m_LEDStates->DiagnosticPattern(FMSData::GetInstance()->GetAllianceColor(), backintake, frontintake, feeder, launcher, placerin, placermid, placerout);
+        StateMgr *taleStateManager = MechanismConfigMgr::GetInstance()->GetCurrentConfig()->GetMechanism(MechanismTypes::DRAGON_TALE);
+        StateMgr *intakeStateManager = MechanismConfigMgr::GetInstance()->GetCurrentConfig()->GetMechanism(MechanismTypes::INTAKE_MANAGER);
+        auto taleMgr = taleStateManager != nullptr ? dynamic_cast<DragonTale *>(taleStateManager) : nullptr;
+        auto intakeMgr = intakeStateManager != nullptr ? dynamic_cast<IntakeManager *>(intakeStateManager) : nullptr;
+
+        if (taleMgr != nullptr && intakeMgr != nullptr)
+        {
+            bool coralInSensor = taleMgr->GetCoralInSensorState();
+            bool coralOutSensor = taleMgr->GetCoralOutSensorState();
+            bool algaeSensor = taleMgr->GetAlgaeSensorState();
+            bool intsakeSensor = intakeMgr->GetIntakeSensorState();
+            m_LEDStates->DiagnosticPattern(FMSData::GetInstance()->GetAllianceColor(), coralInSensor, coralOutSensor, algaeSensor, intsakeSensor);
```

Comment:
Where is the diagnostic LEDs for things like Limelights, DragonQuest, Drive State, Heading State, etc.?